Content.Client/Construction/UI/ConstructionMenuPresenter.cs

```diff
@@ -201,12 +214,79 @@ private void OnViewPopulateRecipes(object? sender, (string search, string catago
 
             recipes.Sort((a, b) => string.Compare(a.Name, b.Name, StringComparison.InvariantCulture));
 
-            foreach (var recipe in recipes)
+            var recipesList = _constructionView.Recipes;
+            recipesList.Clear();
+
+            var recipesGrid = _constructionView.RecipesGrid;
+            recipesGrid.RemoveAllChildren();
+
+            _constructionView.RecipesGridScrollContainer.Visible = _constructionView.GridViewButtonPressed;
+            _constructionView.Recipes.Visible = !_constructionView.GridViewButtonPressed;
+
+            if (_constructionView.GridViewButtonPressed)
             {
-                recipesList.Add(GetItem(recipe, recipesList));
+                foreach (var recipe in recipes)
+                {
+                    var item = GetItem(recipe, recipesList);
+
+                    var itemButton = new TextureButton
+                    {
+                        TextureNormal = item.Icon,
+                        VerticalAlignment = Control.VAlignment.Center,
+                        Name = recipe.Name,
+                        ToolTip = recipe.Name,
+                        Scale = new Vector2(1.35f),
+                        ToggleMode = true,
+                    };
+                    var itemButtonPanelContainer = new PanelContainer
+                    {
+                        PanelOverride = new StyleBoxFlat { BackgroundColor = StyleNano.ButtonColorDefault },
+                        Children = { itemButton },
+                    };
+
+                    itemButton.OnToggled += buttonToggledEventArgs =>
+                    {
+                        if (_selected is null)
+                            SelectGridButton(itemButton, buttonToggledEventArgs.Pressed);
+                        else
+                        {
+                            SelectGridButton(itemButton, true);
+                            if (_recipeButtons.TryGetValue(_selected.Name, out var oldButton))
+                            {
+                                oldButton.Pressed = false;
+                                SelectGridButton(oldButton, false);
+                                if (_selected == recipe)
+                                {
+                                    OnGridViewRecipeSelected(this, null);
+                                    return;
+                                }
+                            }
+                        }
+                        OnGridViewRecipeSelected(this, buttonToggledEventArgs.Pressed ? recipe : null);
```

Comment:
This section is a bit convoluted. Try this:

```
SelectGridButton(itemButton, buttonToggledEventArgs.Pressed);

if (buttonToggledEventArgs.Pressed &&
    _selected != null &&
    _recipeButtons.TryGetValue(_selected.Name, out var oldButton))
{
    oldButton.Pressed = false;
    SelectGridButton(oldButton, false);
}

OnGridViewRecipeSelected(this, buttonToggledEventArgs.Pressed ? recipe : null);
```

Reply:
Nice, but when you exit and re-enter grid view with a button already selected, it forgets it's selected and you can't unselect it. Thankfully, I was able to fix this by adding `itemButton.Pressed = _selected == recipe` in the populating loop. Changes applied.